mypy/nodes.py

```diff
@@ -334,7 +336,16 @@ def __init__(self, arguments: List[Argument], body: 'Block',
         arg_kinds = [arg.kind for arg in self.arguments]
         self.max_pos = arg_kinds.count(ARG_POS) + arg_kinds.count(ARG_OPT)
         self.body = body
-        self.type = typ
+        if typ is not None:
+            self.type = typ
+        elif Options.get_options().implicit_any:
+            from mypy.types import (UnboundType, CallableType)
+            self.type = CallableType([UnboundType('Any', []) for _ in arguments],
```

Comment:
Use `AnyType` instead of `UnboundType` here, since `UnboundType('Any', ...)` works as expected only if there is an explicit `from typing import Any` in the file. For example, if there is just `import typing` or no import at all this won't get bound during semantic analysis.